### PR TITLE
Add GraphQL reference

### PR DESCRIPTION
**Pull request by dshevtsov**
## Description

Add the GraphQL reference generated by SpectaQL. The reference documentation is a single file `static/graphql-api/index.html`. It is embedded using the frameSrc feature in the `src/pages/graphql/reference.md` file.
The SpectaQL project is located in the `spectaql` directory. After any changes in the spectaql project, for instance, after updating the shema.json file, regenerate the reference documentation as the following:

```
yarn build:spectaql
```

Preview:
* The generated page: https://adobedocs.github.io/commerce-webapi/graphql-api
* The DevSite page where the generated page is embedded: https://adobedocs.github.io/commerce-webapi/graphql/reference/. ~DevSite does not allow to use of static pages at the moment. The DevSite team announced that it should be fixed very soon.~

### Pre-merge actions

When ready to merge:

- [x] In `src/pages/graphql/reference.md`, change `frameSrc: https://adobedocs.github.io/commerce-webapi/graphql-api` to `frameSrc: https://developer.adobe.com/commerce/webapi/graphql-api`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dshevtsov** [comment]
The PR validation check fails on the content unrelated on this PR.

File: package.json [hunk 1]
```diff
@@ -7,14 +7,15 @@
     "url": "https://github.com/AdobeDocs/commerce-webapi"
   },
   "dependencies": {
-    "@adobe/gatsby-theme-aio": "4.6.6",
+    "@adobe/gatsby-theme-aio": "^4.7.9",
```
Comment (jeff-matthews) [inline package.json:10]: We haven't bumped the theme yeet in any other repos. I think @bdenham mentioned that this version throws some JS console errors. I recommend backing this out unless it's necessary for this PR.

File: package.json [hunk 2]
```diff
@@ -24,7 +25,9 @@
     "build": "gatsby build",
     "serve": "gatsby serve",
     "clean": "gatsby clean",
-    "test:links": "remark src/pages --ignore-pattern src/pages/graphql,src/pages/_includes --quiet --frail"
+    "test:links": "remark src/pages --quiet --frail",
```
Comment (jeff-matthews) [inline package.json:28]: If you remove the ignore pattern, the link checker will fail on false positives. See the description in https://github.com/AdobeDocs/commerce-webapi/pull/57.

**Review by jeff-matthews — COMMENTED** [review]
In addition to my comments below, I'm also curious how we expect users to access these docs since I didn't see any navigation updates.

File: package.json (hunk 1 above)
Comment (dshevtsov) [reply package.json:10]: Just wanted to enable this fix: https://jira.corp.adobe.com/browse/DEVSITE-600

Comment (jeff-matthews) [reply package.json:10]: Ah, that's necessary for this PR. Thanks for confirming.

Comment (jeff-matthews) [reply package.json]: @dshevtsov, if you need the fix for frameSrc then I think we must bump the version. But I would pin it to 4.7.9.

Comment (dshevtsov) [reply package.json]: We still can use the absolute link until all the tests are complete to be sure 4.7.9 won't do any harm.

**Comment by dshevtsov** [comment]
> In addition to my comments below, I'm also curious how we expect users to access these docs since I didn't see any navigation updates.

I assume this will be done in future PRs together with other related content updates by the content owner of the GraphQL Guide.

**Comment by jeff-matthews** [comment]
@keharper, what do you think?

**Comment by keharper** [comment]
Maybe we should add an item in the left navigation under schema and before Cart that says "Reference" and points to the autogenerated docs

**Comment by dshevtsov** [comment]
> > In addition to my comments below, I'm also curious how we expect users to access these docs since I didn't see any navigation updates.
> 
> I assume this will be done in future PRs together with other related content updates by the content owner of the GraphQL Guide.

With this I meant that the guide owner should add appropriate navigation in later PRs according to their info structure vision. The purpose of this PR is to implement the GraphQL reference infrastructure.

**Comment by jeff-matthews** [comment]
Got it.

There's one more thing that I'm curious about. The staging build shows the docs without the devsite theme header, footer, and side nav. That's different than the REST API docs, which include header, footer, and side nav. Is that expected? I would assume that since we're using the same implementation method for both REST and GraphQL that they would both render with header, footer, and side nav.

**Comment by jhadobe** [comment]
- Is it possible to make the code samples dynamic width? They get cutoff and require horizontal scrolling if the page is too narrow.
- I'm also noticing the staged pages lag a bit on opening/closing, I assume due to the size of the page.

**Comment by dshevtsov** [comment]
@jeff-matthews, you're probably viewing the generated page itself https://adobedocs.github.io/commerce-webapi/graphql-api.
There is a second link in the PR description that demonstrates the DevSite's page where that generated page is included, with header, footer, and side navigation.

**Review by jeff-matthews — APPROVED** [review]
Aha! Thanks @dshevtsov. I don't see any reason not to merge it.

**Pull request merged**